File: studyq-backend/routes/content_routes.py
from flask import Blueprint
from flask_jwt_extended import jwt_required, get_jwt_identity

from models.document import Document
from services.ai_service import generate_flashcards, generate_notes
from utils.helpers import success_response, error_response

content_bp = Blueprint("content", __name__)


@content_bp.route("/notes/<int:doc_id>", methods=["GET"])
@jwt_required()
def get_notes(doc_id):
    user_id = int(get_jwt_identity())
    doc = Document.query.filter_by(id=doc_id, user_id=user_id).first()
    if not doc:
        return error_response("Document not found.", 404)
    return success_response({"notes": doc.notes or "", "file_name": doc.file_name})


@content_bp.route("/notes/<int:doc_id>/regenerate", methods=["POST"])
@jwt_required()
def regenerate_notes(doc_id):
    user_id = int(get_jwt_identity())
    doc = Document.query.filter_by(id=doc_id, user_id=user_id).first()
    if not doc:
        return error_response("Document not found.", 404)
    try:
        from models import db
        doc.notes = generate_notes(doc.extracted_text)
        db.session.commit()
        return success_response({"notes": doc.notes, "file_name": doc.file_name}, "Notes regenerated.")
    except Exception as e:
        return error_response(str(e), 500)


@content_bp.route("/flashcards/generate/<int:doc_id>", methods=["POST"])
@jwt_required()
def generate_flashcards_route(doc_id):
    user_id = int(get_jwt_identity())
    doc = Document.query.filter_by(id=doc_id, user_id=user_id).first()
    if not doc:
        return error_response("Document not found.", 404)
    try:
        cards = generate_flashcards(doc.extracted_text)
        return success_response(
            {"flashcards": cards, "file_name": doc.file_name},
            "Flashcards generated successfully.",
            201,
        )
    except ValueError as e:
        return error_response(str(e), 422)
    except RuntimeError as e:
        return error_response(str(e), 500)

File: studyq-backend/routes/contentRoutes.js
const express = require('express');
const Document = require('../models/Document');
const { generateNotes, generateFlashcards } = require('../services/aiService');
const { successResponse, errorResponse } = require('../utils/helpers');
const authMiddleware = require('../middleware/auth');

const router = express.Router();

router.get('/notes/:doc_id', authMiddleware, async (req, res) => {
  try {
    const doc = await Document.findOne({ _id: req.params.doc_id, user_id: req.user.id });
    if (!doc) {
      return errorResponse(res, "Document not found.", 404);
    }
    return successResponse(res, { notes: doc.notes || "", file_name: doc.file_name });
  } catch (err) {
    console.error(err);
    return errorResponse(res, "Server error fetching notes.", 500);
  }
});

router.post('/notes/:doc_id/regenerate', authMiddleware, async (req, res) => {
  try {
    const doc = await Document.findOne({ _id: req.params.doc_id, user_id: req.user.id });
    if (!doc) {
      return errorResponse(res, "Document not found.", 404);
    }
    try {
      doc.notes = await generateNotes(doc.extracted_text);
      await doc.save();
      return successResponse(res, { notes: doc.notes, file_name: doc.file_name }, "Notes regenerated.");
    } catch (err) {
      return errorResponse(res, err.message, 500);
    }
  } catch (err) {
    console.error(err);
    return errorResponse(res, "Server error regenerating notes.", 500);
  }
});

router.post('/flashcards/generate/:doc_id', authMiddleware, async (req, res) => {
  try {
    const doc = await Document.findOne({ _id: req.params.doc_id, user_id: req.user.id });
    if (!doc) {
      return errorResponse(res, "Document not found.", 404);
    }
    try {
      const cards = await generateFlashcards(doc.extracted_text);
      return successResponse(res, { flashcards: cards, file_name: doc.file_name }, "Flashcards generated successfully.", 201);
    } catch (err) {
      return errorResponse(res, err.message, 422);
    }
  } catch (err) {
    console.error(err);
    return errorResponse(res, "Server error generating flashcards.", 500);
  }
});

module.exports = router;
